Python/Product/VSInterpreters/DerivedInterpreterFactory.cs

```diff
@@ -266,5 +267,81 @@ InterpreterFactoryCreationOptions options
             }
             return base.GetIsCurrentReason(culture);
         }
+
+        public static IPythonInterpreterFactory FindBaseInterpreterFromVirtualEnv(
+            string prefixPath,
+            string libPath,
+            IInterpreterOptionsService service
+        ) {
+            string basePath = GetOrigPrefixPath(prefixPath, libPath);
+
+            if (Directory.Exists(basePath)) {
+                return service.Interpreters.FirstOrDefault(interp =>
+                    CommonUtils.IsSamePath(interp.Configuration.PrefixPath, basePath)
+                );
+            }
+            return null;
+        }
+
+        public static string GetOrigPrefixPath(string prefixPath, string libPath = null) {
+            string basePath = null;
+
+            if (!Directory.Exists(prefixPath)) {
+                return null;
+            }
+
+            var cfgFile = Path.Combine(prefixPath, "pyvenv.cfg");
+            if (File.Exists(cfgFile)) {
+                try {
+                    var lines = File.ReadAllLines(cfgFile);
+                    basePath = lines
+                        .Select(line => Regex.Match(line, "^home *= *(?<path>.+)$", RegexOptions.IgnoreCase))

```

Comment:
Strictly it should be `"[ \t]+"`, according to the CPython sources. I'm okay with being more forgiving here though.